functions.php

```diff
@@ -52,3 +52,21 @@ function sanitization_example_function( $post_id ) {
 function escaping_example_function( $post_id ) {
 	echo esc_html( get_post_meta( $post_id, 'some_key' ) );
 }
+
+/**
+ * Gets paid posts if the user has a subscription.
+ *
+ * @return array
+ */
+function optimization_example_function() {
+	$paid_posts = get_posts( array(
+		'post_status' => 'publish',
+		'post_type' => 'paid',
+	) );
+
+	if ( current_user_can( 'read_paid_posts' ) ) {
```

Comment:
Why not check this before running the query? If the user doesn't have access, there's no reason to run it.